.github/workflows/docker-merge.yml

```diff
@@ -6,6 +6,10 @@ on:
   push:
     branches:
       - 'trunk'
+  # to test pass/failure before a merge
+  pull_request:
```

Comment:
I understand why this was added in, but I am not sure that we want this here. 

I am not 100% sure what the reason is that we have different workflows for pull, push, and merge, but my gut reaction is we should not modify when they are run and instead synchronize these workflows (if they've gotten out of sync). 

@HalosGhost would be good to get your insight here if possible.

@rockett-m I think this should be removed from this commit for now

Reply:
I need to remind myself of why we had this separation; it was definitely intentional, but I cannot recall what the purpose was. Let me dig into this and see if I can recover an answer.

Reply:
If we don't run merge before the merge takes place then we can have future CI/CD breakage

Reply:
That's because, in this case, `pull` somehow didn't fail despite `activate-venv.sh` not being copied to the container. `merge` did fail; i.e., the two actions have a gap in functionality. *That* is what should be fixed; not having `merge` run on pull-requests.

Reply:
ACK - makes sense
doing a comparison between the docker-pull and docker-merge files, I see a point of concern

> Limiting fetch depth in docker-pull (to only the most recent 2 commits) - necessary?
```
    ########################
    # CI Setup             #
    ########################
    - name: Checkout code
      uses: actions/checkout@v3
        with:                                        # only in docker-pull
          fetch-depth: "2"                           # only in docker-pull
```
https://git-scm.com/docs/git-fetch

Reply:
Note: I'm comfortable getting this merged as a fast-follow without having solved the gap between `pull` and `merge`, but this PR should not add running `merge` on PRs.

Reply:
I wouldn't think the workflow doing a shallow clone would cause any issue here (it will still have all the files present at both of those commits)—I would still expect the docker build to fail if the file isn't copied to the container as-required.

Reply:
TACK - this was not the issue